pkg/cluster/k8sres.go

```diff
@@ -485,13 +486,14 @@ func generatePodTemplate(
 
 	template := v1.PodTemplateSpec{
 		ObjectMeta: metav1.ObjectMeta{
-			Labels:    labels,
-			Namespace: namespace,
+			Labels:      labels,
+			Namespace:   namespace,
+			Annotations: annotations,
 		},
 		Spec: podSpec,
 	}
 	if kubeIAMRole != "" {
-		template.Annotations = map[string]string{constants.KubeIAmAnnotation: kubeIAMRole}
+		template.Annotations[constants.KubeIAmAnnotation] = kubeIAMRole
```

Comment:
This should check whether Annotations is nil since generatePodAnnotations may return nil